sopel/config/types.py

```diff
@@ -47,6 +47,13 @@ class StaticSection(object):
 
     This class is intended to be subclassed and customized with added
     attributes containing :class:`BaseValidated`-based objects.
+
+    .. note::
+
+        Subclasses of ``StaticSection`` are named ``PluginnameSection`` by
+        convention—i.e. the plugin's name in strict CamelCase, with the suffix
+        ``Section``. However, any desired class name should work just fine.
```

Comment:
Tagging on to @Exirel's apt suggestion on the first line of this block and trying to show an alternative:

```suggestion
        By convention, subclasses of ``StaticSection`` are named with the
        plugin's name in CamelCase, plus the suffix ``Section``. For example, a
        plugin named ``editor`` might name its subclass ``EditorSection``.
        However, any desired class name should work just fine.
```

Perhaps I should also add a note somewhere in `plugin.rst` about the conventional naming scheme for plugins (all lowercase, usually one word, or snake_case if necessary), since I'm already on the subject? Hmm…